File: Problems/USACO/chapter1/sec1.2./friday.cpp
/*
ID: grgurin1
TASK: friday
LANG: C++                 
*/
#include <iostream>
#include <fstream>
#include <string>

using namespace std;
int months[12] = {31, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30};
int ans[7] = {0};
int years;


int main() {
    ifstream fin("friday.in");
    ofstream fout("friday.out");

    fin >> years;

    int day = 4;
    for(int y = 1900; y<1900+years; y++)
    for(int m = 0; m<12; m++) {
        
        if (((y%100 == 0 && y%400 == 0) || (y%100 != 0 && y%4 == 0)) && m == 2) {
            day = (day + months[m] + 1)%7;
        } else {
            day = (day + months[m])%7;
        }
        ans[day] += 1;
    }

    for(int i = 0; i<6; i++) fout << ans[i] << " ";
    fout << ans[6] << "\n";
}

File: Problems/USACO/chapter1/sec1.2./friday.py
"""
ID: grgurin1
LANG: PYTHON3
TASK: friday
"""

fin = open('friday.in', 'r')

n = int(fin.readline()) + 1900

year = 1900
month = 1
last_day = 0
ans = [0]*7


while year < n:
    for _ in range(12):
        ans[last_day] += 1
        diff = 31

        # Korištenje operacija
        if month == 2:
            diff = 28
            if year % 4 == 0 and year % 100 != 0:
                diff += 1
            elif year % 400 == 0:
                diff += 1
        elif month in (4, 6, 9, 11):
            diff = 30
        
        last_day = (diff+last_day) % 7
        
        month += 1
    month = 1
    year += 1



fout = open('friday.out', 'w')
fout.write(str(ans).strip(']').strip('[').replace(',', '')+'\n')
fout.close()


